Source/WebCore/page/EventHandler.cpp

```diff
@@ -4614,17 +4619,16 @@ bool EventHandler::startKeyboardScrollAnimationOnRenderBoxAndItsAncestors(Scroll
     return false;
 }
 
-bool EventHandler::startKeyboardScrollAnimationOnPlugin(ScrollDirection direction, ScrollGranularity granularity, RenderEmbeddedObject& plugin, bool isKeyRepeat)
+bool EventHandler::startKeyboardScrollAnimationOnPlugin(ScrollDirection direction, ScrollGranularity granularity, RenderEmbeddedObject& pluginRenderer, bool isKeyRepeat)
 {
-    if (!plugin.usesAsyncScrolling())
-        return false;
-    ScrollingNodeID scroller = plugin.scrollingNodeID();
-    ScrollableArea* scrollableArea = m_frame->view()->scrollableAreaForScrollingNodeID(scroller);
+    auto* scrollableArea = pluginRenderer.scrollableArea();
```

Comment:
```suggestion
    RefPtr scrollableArea = pluginRenderer.scrollableArea();
```

Reply:
ScrollableArea is not refcounted (but this could be a `CheckedPtr`)